lib/reline/line_editor.rb

```diff
@@ -1148,6 +1181,31 @@ def input_key(key)
     modified
   end
 
+  def save_old_buffer
+    @old_buffer_of_lines = @buffer_of_lines.dup
+    @old_byte_pointer = @byte_pointer.dup
+    @old_line_index = @line_index.dup
+  end
+
+  MAX_PAST_LINES = 100
+  def save_past_lines
+    if @old_buffer_of_lines != @buffer_of_lines
+      if !@using_delete_command && @buffer_of_lines == @past_lines.last.first
```

Comment:
We need `@past_lines.empty?` check.
Reline raises no method error with input `undo x delete undo x`

Reply:
It seems better to initialize with `@past_lines = []` and then check if it's empty. I fixed.